python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition_evaluator.py

```diff
@@ -102,13 +103,16 @@ def prefetch(self) -> None:
         self.logger.info("Done prefetching asset records.")
 
     def evaluate(self) -> Tuple[Sequence[AutomationResult], Sequence[EntitySubset[EntityKey]]]:
+        return asyncio.run(self.async_evaluate())
+
+    async def async_evaluate(
+        self,
+    ) -> Tuple[Sequence[AutomationResult], Sequence[EntitySubset[EntityKey]]]:
         self.prefetch()
         num_conditions = len(self.entity_keys)
         num_evaluated = 0
-        for entity_key in self.asset_graph.toposorted_entity_keys:
-            if entity_key not in self.entity_keys:
-                continue
 
+        async def _evaluate_entity_async(entity_key: EntityKey) -> int:
             self.logger.debug(
                 f"Evaluating {entity_key.to_user_string()} ({num_evaluated+1}/{num_conditions})"
```

Comment:
looking at this a bit closer, this num_evaluated number isn't going to be quite right, as it'll be the same for all assets within the same topological level.

this is really just a debugging helper, so it's not a hard requirement that we nail this, but I think the easier thing here would be to do enumerate the entity keys when creating the coroutines on line 142, and then pass that in as an offset, i.e.:

```python
num_evaluated = 0

async def _evaluate_entity_async(entity_key: EntityKey, offset: int) -> None:
   f"evaluating ... {num_evaluated+offset}/{num_conditions}"

for topo_level in ...:
    coroutines = [
        _eval_entity_async(entity_key, offset) for ... in enumerate(topo_level)
    ]
    gathered = ...
    num_evaluated += len(coroutines)
```

this at least lets us assign a unique number to each thing we evaluate